src/token/token.handler.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@nestjs/common';
+import { OnCommand } from 'discord-nestjs';
+import { Message } from 'discord.js';
+import { TokenService } from './token.service';
+
+@Injectable()
+export class TokenHandler {
+  constructor(private readonly tokenService: TokenService) {}
+
+  @OnCommand({ name: 'create-token' })
+  async createToken(message: Message) {
+    if (message.member.roles.cache.some((role) => role.name === 'Moderators')) {
```

Comment:
Maybe also the channel should be checked, so this is not run in public channels? or they get a reply with a DM?

Reply:
Moved it to a DM-Message, because I'm not a fan of hardcoding a channel

Reply:
ok great 👍